www/src/pages/table/forms/TransformationForm.tsx

```diff
@@ -0,0 +1,289 @@
+                  }
+                </>
+              );
+            } else {
+              return transformed_value;
+            }
+          },
+        },
+      ];
+
+      const onExecute = async () => {
+        const transformationPayload: Transformation = {
+          id: -1,
+          tableId: table.id,
+          type: form.getFieldValue("type"),
+          code: form.getFieldValue("code"),
+          mode: "restrictedpython",
+          datapath: form.getFieldValue("datapath"),
+          outputpath: form.getFieldValue("outputpath"),
+          tolerance: form.getFieldValue("tolerance"),
+          rows: form.getFieldValue("rows"),
+        };
+
+        let response = await transformationStore.testTransformation(
+          transformationPayload
+        );
+        setResult(response);
+      };
+
+      return (
+        <Form form={form}>
+          <Row gutter={16} justify="start">
+            <Col span={4}>
+              <Form.Item name="type" label="type" rules={[{ required: true }]}>
+                <Select placeholder="Transformation Type" allowClear={true}>
+                  <Select.Option value="map">Map</Select.Option>
+                  <Select.Option value="filter">Filter</Select.Option>
+                  <Select.Option value="split">Split</Select.Option>
+                  <Select.Option value="concatenate">Concatenate</Select.Option>
+                </Select>
+              </Form.Item>
+            </Col>
+            <Col span={10}>
+              <Form.Item
+                name="datapath"
+                label="datapath"
+                rules={[{ required: true }]}
+              >
+                <Select
+                  mode="tags"
+                  allowClear={true}
+                  style={{
+                    width: "100%",
+                  }}
+                  placeholder="Please select columns"
+                  options={table.columns.map((column) => ({
+                    label: column,
+                    value: column,
+                  }))}
+                />
+              </Form.Item>
+            </Col>
+            <Space></Space>
+            <Col span={10}>
+              <Form.Item
+                name="outputpath"
+                label="outputpath"
+                rules={[{ required: true }]}
+              >
+                <Select
+                  mode="tags"
+                  allowClear={true}
+                  style={{
+                    width: "100%",
+                  }}
+                  options={table.columns.map((column) => ({
+                    label: column,
+                    value: column,
+                  }))}
+                  placeholder="Please select columns"
+                />
+              </Form.Item>
+            </Col>
+          </Row>
+          <Row justify="center" align="top">
+            <Col span={24}>
+              <Form.Item
+                name="code"
+                label="Language : Restricted Python"
```

Comment:
extra space before : is it for aesthetic purpose?

Reply:
updated it